test/insert.js

```diff
@@ -0,0 +1,58 @@
+'use strict';
+
+var S = require('..');
+
+var eq = require('./internal/eq');
+var jsc = require('jsverify');
+
+
+test('insert', function() {
+
+  eq(typeof S.insert, 'function');
+  eq(S.insert.length, 3);
+  eq(S.insert.toString(), 'insert :: String -> a -> StrMap a -> StrMap a');
+
+  function testBasic(key, val, strMap, strMapResult) {
+    eq(S.insert(key, val, strMap), strMapResult);
+  }
+
+  function unique(xs) {
+    if (xs.length === 0) return [];
+
+    var sorted = S.sort(xs);
+    var x = sorted[0];
+    return sorted.slice(1).reduce(function(acc, x) {
+      return S.equals(x, acc.prev) ? acc : {xs: S.append(x, acc.xs), prev: x};
+    }, {xs: [x], prev: x}).xs;
+  }
+
+  function testKeysOfInsert(key, val, strMap) {
+    eq(
+      unique(S.sort(S.concat(S.keys(strMap), [key]))),
+      S.sort(S.keys(S.insert(key, val, strMap)))
+    );
+  }
+
+  function testDoubleInsert() {
+    jsc.assert(jsc.forall(jsc.string, jsc.number, jsc.dict(jsc.number), function(key, val, map) {
+      var insert = S.insert(key, val);
+      return S.equals(insert(insert(map)), insert(map));
+    }), {tests: 1000});
+  }
+
+  function testInsert(key, val, strMap, strMapResult) {
+    testBasic(key, val, strMap, strMapResult);
+    testKeysOfInsert(key, val, strMap);
+    testDoubleInsert();
```

Comment:
This means we test the property 1000 times each time we call `testInsert`, which seems excessive. ;)

Testing the property 1000 times rather than 8000 times may prevent the tests from timing out on CircleCI.

Reply:
Silly mistake on my part to call `testDoubleInsert` on every call to `testInsert`

Reply:
No worries. :)